docs/faq.md

```diff
@@ -38,7 +38,94 @@ For nextpnr we are using the following terminology.
 Adding new architectures to nextpnr
 -----------------------------------
 
-TBD
+### Implementing new architectures
+
+Each nextpnr architecture must implement the *nextpnr architecture API*.
+See [archapi.md](archapi.md) for a complete reference of the architecture API.
+
+### Delay Estimates
+
+Each architecture must implement a `estimateDelay()` method that estimates the expected delay for a path from given `src` to `dst` wires.
+*It is very important that this method slightly overestimates the expected delay.* Furthermore, it should overestimate the expected delay
+by a slightly larger margin for longer paths than for shorter paths. Otherwise there will be performance issues with the router.
+
+The delays estimates returned by that method should also be as fine-grain as possible. It definitely pays off to spend some time improving the `estimateDelay()`
+for your architecture once implementing small designs work.
+
```

Comment:
It would be good to describe the difference between `predictDelay` and `estimateDelay` -- my understanding is the former is used in placement, the latter in routing. I would recommend changing their names, as they are too similar, perhaps: `fastEstimateDelay()` and `slowEstimateDelay()`?
Again, not critical for this PR, but dropping this comment here before I forget about it...

Reply:
I was thinking maybe `predictDelay() -> estimateArcDelay()`.

Reply:
Arc is one of the words on our nomenclature list with exactly this meaning. https://github.com/YosysHQ/nextpnr/blob/master/docs/faq.md If the name clash is a problem then we should change our nomenclature. In either case I think it would be a good idea to use whatever word we assign to this concept.

Reply:
Sorry, I'm mistaking Arc for "Pip". `estimateArcDelay` is good.